### PR TITLE
Issue #4749: delete deprecated classes

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4749
delete deprecated classes AbstractComplexityCheck, AbstractIllegalCheck, AbstractIllegalMethodCheck, AbstractNestedDepthCheck

### CODE REVIEW

**Comment by romani** [comment]
@Nimfadora , wercker is failed, please update wercker.yml to comment out validation over sevntu.
Sevntu project need to be updated, will be done in separate issue

**Comment by rnveach** [comment]
@romani I am creating the PR for sevntu.

**Comment by rnveach** [comment]
Fix was merged, restarting wercker

**Comment by rnveach** [comment]
Still failed, `checkstyle's sevntu` uses sevntu release.
We either have to release a new version of sevntu for fix or disable wercker until next sevntu release.

File: pom.xml [hunk 1]
```diff
@@ -1865,10 +1859,6 @@
               <targetTests>
                 <param>com.puppycrawl.tools.checkstyle.checks.metrics.*</param>
               </targetTests>
-              <excludedClasses>
-                <!-- deprecated class -->
-                <param>com.puppycrawl.tools.checkstyle.checks.metrics.AbstractComplexityCheck</param>
-              </excludedClasses>
```
Comment (Vladlis) [inline pom.xml]: Please also remove exclusions from cobertura

**Comment by romani** [comment]
@Nimfadora , please rebase on latest code, I disabled sevntu for you.

**Comment by Nimfadora** [comment]
@romani done

**Comment by romani** [comment]
@Nimfadora , please read review comments and reply to each of it as done.
DO NOT SKIP ANY !!!!

you are missing required updates - https://github.com/checkstyle/checkstyle/pull/4767#pullrequestreview-51031372.

before inviting to review, please make sure that all codereview points are  commented as "done"

File: pom.xml (hunk 1 above)
Comment (Nimfadora) [inline pom.xml]: fixed

**Pull request merged**